front/src/modules/pathfinding/utils.ts

```diff
@@ -33,9 +36,38 @@ export const formatSuggestedOperationalPoints = (
     offsetOnTrack: op.part.position,
     track: op.part.track,
     positionOnPath: op.position,
-    coordinates: getPointCoordinates(geometry, pathLength, op.position),
+    coordinates:
+      (geometry !== undefined &&
+        pathLength !== undefined &&
+        getPointCoordinates(geometry, pathLength, op.position)) ||
+      undefined,
   }));
 
+export const addTrackNamesToSuggestedOperationalPoints = async (
+  operationalPoints: SuggestedOP[],
+  getTrackSectionsByIds: (requestedTrackIds: string[]) => Promise<InfraObjectWithGeometry[]>
+): Promise<SuggestedOP[]> => {
+  // Extract track IDs to fetch track names
+  const trackIds = uniq(operationalPoints.map((op) => op.track));
+  const trackSections = await getTrackSectionsByIds(trackIds);
+
+  return operationalPoints.map((op) => {
+    const associatedTrackSection = trackSections.find(
```

Comment:
Maybe trackSections can be a Map instead of an array ? What do you think ?

Reply:
Honestly I just duplicated the old code for this, but sure!